monkey/monkey_island/cc/ui/src/services/AuthService.js

```diff
@@ -109,12 +147,32 @@ export default class AuthService {
           res.clone().json().then(res_json => {
             console.log('Got 401 from server while trying to authFetch: ' + JSON.stringify(res_json));
           });
-          this._removeToken();
+          this._removeToken(this.AUTH_TOKEN_NAME_IN_LOCALSTORAGE);
         }
         return res;
       })
   };
 
+  _fetchTokenPairFromRefreshToken(url, options = {}) {
+    const headers = {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    };
+
+    if (Object.prototype.hasOwnProperty.call(options, 'headers')) {
+      for (let header in headers) {
+        options['headers'][header] = headers[header];
+      }
+    } else {
+      options['headers'] = headers;
+    }
```

Comment:
Why is this needed?

Reply:
To set up the options which may go away.